qlib/contrib/data/dataset.py

```diff
@@ -130,6 +132,14 @@ def __init__(
         input_size=None,
         **kwargs,
     ):
+        if horizon == 0:
+            # Try to guess horizon
+            if isinstance(handler, dict):
+                handler = init_instance_by_config(handler)
+            elif isinstance(handler, str):  # pickled handler
```

Comment:
Could we write it like this?
`if isinstance(handler, (dict, str))`

Reply:
Done